cmd/k8s-operator/ingress-for-pg.go

```diff
@@ -610,21 +596,38 @@ func (a *IngressPGReconciler) updateTailscaledConfigWithServices(ctx context.Con
 	}
 
 	for _, secret := range secrets.Items {
+		var updated bool
 		for fileName, confB := range secret.Data {
 			var conf ipn.ConfigVAlpha
 			if err := json.Unmarshal(confB, &conf); err != nil {
 				return fmt.Errorf("error unmarshalling ProxyGroup config: %w", err)
 			}
 
-			// Set the new data.
-			conf.AdvertiseServices = servicesToAdvertise
+			// Update the services to advertise if required.
+			idx := slices.Index(conf.AdvertiseServices, serviceName.WithoutPrefix())
```

Comment:
I think the name at this point needs to be with the `svc:` prefix.
We are actually lacking validation in the conffile, but see the validation that would have caught this for `--advertise` command https://github.com/tailscale/tailscale/blob/986daca5eeeffa04bdb184d1ee13f70d04d33ff1/cmd/tailscale/cli/advertise.go#L46